extensions/jms/src/main/java/org/mule/extensions/jms/api/source/DefaultJmsListenerLock.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.jms.api.source;
+
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.message.Error;
+
+import java.util.concurrent.Semaphore;
+
+/**
+ * Default implementation of {@link JmsListenerLock} based in the usage of {@link Semaphore} to support
+ * the capability of locking and unlocking the listener from different {@link Thread threads}
+ *
+ * @since 4.0
+ * @see JmsListenerLock
+ * @see JmsListener
+ */
+public class DefaultJmsListenerLock implements JmsListenerLock {
+
+  private Semaphore semaphore = new Semaphore(0);
```

Comment:
The lock should always be start blocked because since the first message the listener should wait to the flow processing finish.